### ISSUE TITLE
Getting Qt5Gui.dll is missing error when running SEGS for the first time

### ISSUE THREAD

**Issue by Armingleggings** [Segs/Segs#289]
I installed the CoX client and set the server IP to 127.0.0.1. I downloaded SEGS and extracted it. I placed the folder in Program Files (x86). When I run SEGS.exe, I get the above listed error. I can download the DLL, but I suspect something else is wrong. Should it have been included? Was there an assumption made about the user environment that isn't true in my case (a missing library or perhaps something like and SDK)? 

I'm running Windows 7 in my Virtual Machine.

**Comment by Armingleggings** [comment]
My mistake. Is there a guide or walkthrough or setup steps page I missed? If not, we might want to have something to make getting started as easy as possible. 

That said, what if I set it to my local ip address? It would have the same effect as localhost without triggering some of the custom stuff ya?

**Comment by Armingleggings** [comment]
Running authserver gives a series of errors about not finding tables and directories. For example: "no such table: costume" and "couldn't load costume.bin from /data/bin". What am I missing?

**Comment by Armingleggings** [comment]
Following instructions. Had to run dbtool -f because the defaults didnt' seem to be working for some reason. That got rid of the SQL errors, but I'm still getting a few more like "config file is missing location_addr" and such. Going back to see if I missed anything else.

**Comment by Armingleggings** [comment]
Yup, I'm stuck now. It's saying the local_addr value isn't in the config, but it is (unless settings.cfg is a decoy). Also warning me about the pigg files, but I've run the pigtool a few time and it executes just fine.

**Comment by broxen** [comment]
Please ensure that the pigg files are located in the segs folder under `/data/` and likewise that the extracted bin files are located under `/data/bin/`

**Comment by Armingleggings** [comment]
Odd... the pigg files are all in place, but there's no data/bin folder... let me look at those instructions for the piggtool again.

**Comment by Armingleggings** [comment]
OH! It creates bin in the root SEGS folder, but it needs to be then moved to data. Got it.

Out of curiosity, couldn't the piggtool just export to data/bin directly instead of requiring the extra step?

**Comment by Armingleggings** [comment]
I don't see a bin folder anywhere. It says it should be in the SEGS output folder, but I downloaded the stable release (didn't compile). I see and "out" file, but no folder and bin is nowhere in the folder structure according to Windows search.

**Comment by Armingleggings** [comment]
Update: putting SEGS in the Program Files folder creates issues with the installation instructions because of permissions problems. I could have opened an administrative command prompt, but chose to move SEGS to my root folder instead where the permissions are open. Once I did, I had no trouble with the piggtool.

**Comment by Armingleggings** [comment]
still getting a few more like "config file is missing location_addr" and the SQL errors are back.

**Comment by broxen** [comment]
The missing location_addr and listen_addr for the RoamingServer are OK. The SQL errors means that the sq files probably didnt generate properly. Please run dbtool again now that you're in a new folder with perms.

**Comment by broxen** [comment]
Yes output folder does mean the SEGS folder. I think the docs may be missing a step. You need to run binConverter on bin.pigg

**Comment by Armingleggings** [comment]
Well corp... Ok. I'm working on a batch script to automate almost all of this. When I get this all worked out and the batch script is done, how can I submit it for consideration?

**Comment by Armingleggings** [comment]
Can you clarify what has to happen with binconverter? I tried running binConverter data/bin.pigg, but it doesn't seem to do anything. I get the same output about supported type as I do if I run the command with no input.

**Comment by nemerle** [comment]
I think broxen meant `piggtool -x` :)
binConverter is for converting from .bin formats to readable-ish json

**Comment by broxen** [comment]
Ahh yes. I definitely did.

**Comment by Armingleggings** [comment]
I thought that might be the case. Ok, I'll get back to work. 

As for my other question, I think having a batch script to help with the steps will simplify things. Where/how/who should I submit it to when done?

**Comment by nemerle** [comment]
The batch script, You can either do it the github way, ( clone the repository, stick the script into a proper place, do a PR), post it as a gist (gist.github.com), or simply post it somewhere and one of us will integrate it a bit :)

Just keep in mind that the script will be integrated into the SLAV utility ( Segs Launcher And Verifier ) :smile:

**Comment by Armingleggings** [comment]
The goal of the script is JUST to get everything in place and automate the various readme steps. I'll leave it to you fine folks to figure out the details though I am working hard to handle all fail conditions and documentation and user guidance (it's what I do). I've never managed to learn GIT properly so I'll probably cheat and direct message it to someone.  Thanks everyone! I'll comment back if I manage to get it completely working (SEGS and the script that is).

**Comment by broxen** [comment]
Hi Armingleggings. Did you have this batch script to share with the community? I'd like to put it up for further testing and try to get include it in our next release.

**Comment by broxen** [comment]
Thanks team!

**Comment by broxen** [comment]
To add to this, if you're going to create `MapInstances\City_00_01` you might as well create these two other directories ```MapInstances\City_01_01
MapInstances\City_29_01```

Also, as the databases are created with segsadmin user, I would make creating additional accounts optional.
